packages/electron-builder/templates/nsis/uninstaller.nsh

```diff
@@ -33,16 +33,29 @@ Section "un.install"
   ${GetOptions} $R0 "--keep-shortcuts" $R1
   ${if} ${Errors}
     WinShell::UninstAppUserModelId "${APP_ID}"
-    WinShell::UninstShortcut "$startMenuLink"
-    WinShell::UninstShortcut "$desktopLink"
-
-    Delete "$startMenuLink"
-    Delete "$desktopLink"
-
-    # Refresh the desktop
-    System::Call 'shell32::SHChangeNotify(i, i, i, i) v (0x08000000, 0, 0, 0)'
+    WinShell::UninstShortcut "$oldStartMenuLink"
+    WinShell::UninstShortcut "$oldDesktopLink"
+
+    ReadRegStr $R1 SHELL_CONTEXT "${INSTALL_REGISTRY_KEY}" MenuDirectory
+    ${If} $R1 != ""
+      RMDir /r /REBOOTOK "$SMPROGRAMS\$R1"
+    ${Else}
+      Delete "$oldStartMenuLink"
+    ${EndIf}
+    Delete "$oldDesktopLink"
+  ${else}
+    ClearErrors
+    ${GetParameters} $R0
+    ${GetOptions} $R0 "--no-desktop-shortcut" $R1
```

Comment:
@develar , you are right.
I wasn't sure what was best to do here.

We would either enhance the installer to remove any existing oldStartMenuShortcut or let the application owner plan his update (meaning he would have to do 2 subsequent updates).
I selected the second option to not mess the installer with an uninstaller business logic.
But I see that this path leaves garbage: If the owner at the same time changes the shortcut path (shortcut name or menu directory), the desktop icon will never be removed!!!!
The owner would have to control carefully his updates ... That can be frustrating for a lot of people :-(